src/main/scala/io/iohk/ethereum/ledger/BlockImport.scala

```diff
@@ -96,35 +99,51 @@ class BlockImport(
     handleBlockValidationError(error, block)
   }
 
-  private def handleBlockValidationError(error: ValidationBeforeExecError, block: Block): BlockImportResult = error match {
-    case ValidationBeforeExecError(HeaderParentNotFoundError) =>
-      log.debug(s"Block(${block.idTag}) has unknown parent")
-      UnknownParent
+  private def handleBlockValidationError(error: ValidationBeforeExecError, block: Block): BlockImportResult =
+    error match {
+      case ValidationBeforeExecError(HeaderParentNotFoundError) =>
+        log.debug(s"Block(${block.idTag}) has unknown parent")
+        UnknownParent
 
-    case ValidationBeforeExecError(reason) =>
-      log.debug(s"Block(${block.idTag}) failed pre-import validation")
-      BlockImportFailed(reason.toString)
-  }
+      case ValidationBeforeExecError(reason) =>
+        log.debug(s"Block(${block.idTag}) failed pre-import validation")
+        BlockImportFailed(reason.toString)
+    }
 
   private[ledger] def reorganise(
-    block: Block,
-    currentBestBlock: Block,
-    currentTd: BigInt
+      block: Block,
+      currentBestBlock: Block,
+      currentTd: BigInt
   )(implicit blockExecutionContext: ExecutionContext): Future[BlockImportResult] = Future {
-    blockValidation.validateBlockBeforeExecution(block).fold(error => handleBlockValidationError(error, block), _ => {
-      blockQueue.enqueueBlock(block, currentBestBlock.header.number) match {
-        case Some(Leaf(leafHash, leafTd)) if isBetterBranch(block, currentBestBlock, leafTd, currentTd) =>
-          log.debug("Found a better chain, about to reorganise")
-          reorganiseChain(leafHash, leafTd)
-
-        case _ =>
-          BlockEnqueued
-      }
-    })
+    blockValidation
+      .validateBlockBeforeExecution(block)
+      .fold(
+        error => handleBlockValidationError(error, block),
+        _ => {
+          blockQueue.enqueueBlock(block, currentBestBlock.header.number) match {
+            case Some(Leaf(leafHash, leafTd)) if isBetterBranch(block, currentBestBlock, leafTd, currentTd) =>
+              log.debug("Found a better chain, about to reorganise")
+              reorganiseChain(leafHash, leafTd)
+
+            case _ =>
+              BlockEnqueued
+          }
+        }
+      )
   }
 
-  private def isBetterBranch(block: Block, bestBlock: Block, newTd: BigInt, currentTd: BigInt): Boolean =
-    newTd > currentTd || (blockchainConfig.gasTieBreaker && newTd == currentTd && block.header.gasUsed > bestBlock.header.gasUsed)
+  private def isBetterBranch(block: Block, bestBlock: Block, newTd: BigInt, currentTd: BigInt): Boolean = {
```

Comment:
why do we need this ? what about `BranchResolution.compareByCheckpoints / compareByDifficulty` ?